textile/objects-features.textile

```diff
@@ -0,0 +1,145 @@
+** @(RTO6b)@ The expected type of the object can be inferred from the provided @objectId@:
+*** @(RTO6b1)@ Split the @objectId@ (formatted as @type:hash&#64;timestamp@) on the separator @:@ and parse the first part as the type string
+*** @(RTO6b2)@ If the parsed type is @map@, create a zero-value @LiveMap@ per "RTLM4":#RTLM4 in the @ObjectsPool@
+*** @(RTO6b3)@ If the parsed type is @counter@, create a zero-value @LiveCounter@ per "RTLC4":#RTLC4 in the @ObjectsPool@
+
+h3(#livecounter). LiveCounter
+
+* @(RTLC1)@ The @LiveCounter@ extends @LiveObject@
+* @(RTLC2)@ Represents the counter object type for Object IDs of type @counter@
+* @(RTLC3)@ Holds a 64-bit floating-point number as a private @data@
+* @(RTLC4)@ The zero-value @LiveCounter@ is a @LiveCounter@ with @data@ set to 0
+* @(RTLC5)@ @LiveCounter#value@ function:
+** @(RTLC5a)@ Requires the @OBJECT_SUBSCRIBE@ channel mode to be granted per "RTO2":#RTO2
+** @(RTLC5b)@ If the channel is in the @DETACHED@ or @FAILED@ state, the library should indicate an error with code 90001
+** @(RTLC5c)@ Returns the current @data@ value
+* @(RTLC6)@ @LiveCounter@'s internal @data@ can be overridden with the provided @ObjectState@ in the following way:
+** @(RTLC6a)@ Replace the private @siteTimeserials@ of the @LiveCounter@ with the value from @ObjectState.siteTimeserials@
+** @(RTLC6b)@ Set the private flag @createOperationIsMerged@ to @false@
+** @(RTLC6c)@ Set @data@ to the value of @ObjectState.counter.count@, or to 0 if it does not exist
+** @(RTLC6d)@ If @ObjectState.createOp@ is present:
+*** @(RTLC6d1)@ Add @ObjectState.createOp.counter.count@ to @data@, if it exists
+*** @(RTLC6d2)@ Set the private flag @createOperationIsMerged@ to @true@
+
+h3(#livemap). LiveMap
+
+* @(RTLM1)@ The @LiveMap@ extends @LiveObject@
+* @(RTLM2)@ Represents the map object type for Object IDs of type @map@
+* @(RTLM3)@ Holds a @Dict<String, ObjectsMapEntry>@ as a private @data@ map
+* @(RTLM4)@ The zero-value @LiveMap@ is a @LiveMap@ with @data@ set to an empty map
+* @(RTLM5)@ @LiveMap#get@ function:
+** @(RTLM5a)@ Accepts a key of type String
+** @(RTLM5b)@ Requires the @OBJECT_SUBSCRIBE@ channel mode to be granted per "RTO2":#RTO2
+** @(RTLM5c)@ If the channel is in the @DETACHED@ or @FAILED@ state, the library should indicate an error with code 90001
+** @(RTLM5d)@ Returns the value from the current @data@ at the specified key, as follows:
+*** @(RTLM5d1)@ If no @ObjectsMapEntry@ exists at the key, return undefined/null
+*** @(RTLM5d2)@ If a @ObjectsMapEntry@ exists at the key:
+**** @(RTLM5d2a)@ If @ObjectsMapEntry.tombstone@ is @true@, return undefined/null
+**** @(RTLM5d2b)@ If @ObjectsMapEntry.data.boolean@ exists, return it
+**** @(RTLM5d2c)@ If @ObjectsMapEntry.data.bytes@ exists, return it
+**** @(RTLM5d2d)@ If @ObjectsMapEntry.data.number@ exists, return it
+**** @(RTLM5d2e)@ If @ObjectsMapEntry.data.string@ exists, return it
```

Comment:
Per OD5b3, the contents of `string` may be a JSON object or array:

> *** @(OD5b3)@ If @ObjectData.encoding@ is set to "json", the @ObjectData.string@ content is decoded by parsing the string as JSON

But, at least in the JS type declarations and documentation, it seems like `LiveMap.get` cannot return a JSON-like object: https://github.com/ably/ably-js/blob/6f2b7e654921dbda08e178939b1bfd230fd78d4a/ably.d.ts#L2518

Should it be able to?

Reply:
Seems this is addressed by https://github.com/ably/ably-js/pull/2052